src/hotspot/share/utilities/align.hpp

```diff
@@ -72,7 +72,7 @@ constexpr T align_down(T size, A alignment) {
 
 template<typename T, typename A, ENABLE_IF(std::is_integral<T>::value)>
 constexpr T align_up(T size, A alignment) {
-  T adjusted = checked_cast<T>(size + alignment_mask(alignment));
+  T adjusted = checked_cast<T, true>(size + alignment_mask(alignment));
```

Comment:
I guess the checked_cast is needed for sizeof(T) < sizeof(int), but there is still a possible signed overflow UB here.

Reply:
Yeah, looks like align_up has always had that potential overflow problem.
https://bugs.openjdk.org/browse/JDK-8318127